src/ResourceManager/Blueprint/Commands.Blueprint/Common/BlueprintClient.cs

```diff
@@ -105,39 +108,49 @@ public async Task<PSPublishedBlueprint> GetLatestPublishedBlueprintAsync(string
+        public IEnumerable<IEnumerable<PSBlueprintAssignment>> ListBlueprintAssignments(string subscriptionId)
         {
-            var list = new List<PSBlueprintAssignment>();
-            var response = await blueprintManagementClient.Assignments.ListWithHttpMessagesAsync(subscriptionId);
+            var responseList = blueprintManagementClient.Assignments.List(subscriptionId);
 
-            list.AddRange(response.Body.Select(assignment => PSBlueprintAssignment.FromAssignment(assignment, subscriptionId)));
+            yield return responseList.Select(assignment => PSBlueprintAssignment.FromAssignment(assignment, subscriptionId));  
 
-            while (response.Body.NextPageLink != null)
+            while (!string.IsNullOrEmpty(responseList.NextPageLink))
             {
-                response = await blueprintManagementClient.Assignments.ListNextWithHttpMessagesAsync(response.Body.NextPageLink);
-                list.AddRange(response.Body.Select(assignment => PSBlueprintAssignment.FromAssignment(assignment, subscriptionId)));
+                responseList = blueprintManagementClient.Assignments.ListNext(responseList.NextPageLink);
+                yield return responseList.Select(assignment => PSBlueprintAssignment.FromAssignment(assignment, subscriptionId));
             }
-
-            return list;
         }
 
-        public async Task<IEnumerable<PSBlueprint>> ListBlueprintsAsync(string mgName)
+        public IEnumerable<IEnumerable<PSBlueprint>> ListBlueprints(string mgName)
         {
-            var list = new List<PSBlueprint>();
-            var response = await blueprintManagementClient.Blueprints.ListWithHttpMessagesAsync(mgName);
+            var responseList = blueprintManagementClient.Blueprints.List(mgName); 
 
-            list.AddRange(response.Body.Select(bp => PSBlueprint.FromBlueprintModel(bp, mgName)));
+            yield return responseList.Select(bp => PSBlueprint.FromBlueprintModel(bp, mgName));
 
-            while (response.Body.NextPageLink != null)
+            while (!string.IsNullOrEmpty(responseList.NextPageLink))
             {
-                response = await blueprintManagementClient.Blueprints.ListNextWithHttpMessagesAsync(response.Body.NextPageLink);
-                list.AddRange(response.Body.Select(bp => PSBlueprint.FromBlueprintModel(bp, mgName)));
+                responseList = blueprintManagementClient.Blueprints.ListNext(responseList.NextPageLink);
+                yield return responseList.Select(bp => PSBlueprint.FromBlueprintModel(bp, mgName));
+            }    
+        }
+
+        public IEnumerable<IEnumerable<PSBlueprint>> ListBlueprints(List<string> mgList)
+        {
+            foreach (var mgName in mgList)
+            {
+                var listResponse = blueprintManagementClient.Blueprints.List(mgName);
+
+                yield return listResponse.Select(bp => PSBlueprint.FromBlueprintModel(bp, mgName));
```

Comment:
```suggestion
                foreach(var bp in listResponse.Select(bp => PSBlueprint.FromBlueprintModel(bp, mgName)))
                {
                    yield return bp;
                }
```

and then your function can return 
``` C#
public IEnumerable<PSBlueprint> ListBlueprints(List<string> mgList)
```